src/Xml/Document.php

```diff
@@ -196,34 +199,247 @@ protected function getValueCollection(SimpleXMLElement $content, array $matches,
+     * @param  array  $uses
+     * @param  string  $objectName
+     *
+     * @return array
+     */
+    protected function parseValueCollectionMultiLevels(SimpleXMLElement $content, array $uses,$objectName)
+    {
+        $value = [];
+        $result = [];
+        $feature = $content->{$objectName};
+
+        if (!empty($feature)) {
+            foreach ($feature as $key => $contentArray) {
+                foreach ($uses as $use) {
+                    if (strpos($use, '{') !== false) {
+                        $outputSecondLevel = $this->prepareUse($use);
+                        $parent2 = $outputSecondLevel['parent'];
+                        $alias2 = $outputSecondLevel['parent_alias'];
+
+                        $value[$alias2] = $this->parseValueCollectionMultiLevels($contentArray,$outputSecondLevel['array'],$parent2);
+                    } else {
+                        list($name, $as) = strpos($use, '>') !== false ? explode('>', $use, 2) : [$use, $use];
+                        if (preg_match('/^([A-Za-z0-9_\-\.]+)\((.*)\=(.*)\)$/', $name, $matches)) {
+                            if ($name == $as) {
+                                $as = null;
+                            }
+
+                            $item = $this->getSelfMatchingValue($content, $matches, $as);
+
+                            if (is_null($as)) {
+                                $value = array_merge($value, $item);
+                            } else {
+                                Arr::set($value, $as, $item);
+                            }
+                        } else {
+
+                            if ($name == '@') {
+                                $name = null;
+                            }
+                            Arr::set($value, $as, $this->getValue($contentArray, $name));
+                        }
+                    }
+                }
+                $result[] = $value;
             }
         }
+        return $result;
+    }
 
+    /**
+     * Resolve values by collection of one level.
+     *
+     * @param  \SimpleXMLElement  $content
+     * @param  array  $use
+     *
+     * @return array
+     */
+    protected function parseValueCollectionOneLevel(SimpleXMLElement $content, $use)
+    {
+        $value = [];
+
+        list($name, $as) = strpos($use, '>') !== false ? explode('>', $use, 2) : [$use, $use];
+
+        if (preg_match('/^([A-Za-z0-9_\-\.]+)\((.*)\=(.*)\)$/', $name, $matches)) {
+            if ($name == $as) {
+                $as = null;
+            }
+
+            $item = $this->getSelfMatchingValue($content, $matches, $as);
+
+            if (is_null($as)) {
+                $value = array_merge($value, $item);
+            } else {
+                Arr::set($value, $as, $item);
+            }
+        } else {
+            if ($name == '@') {
+                $name = null;
+            }
+            Arr::set($value, $as, $this->getValue($content, $name));
+        }
```

Comment:
I need some tests that cover this use case.

https://coveralls.io/builds/16556955/source?filename=src%2FXml%2FDocument.php

Reply:
Sorry i forget to remove this function, it is just for test something, i will remove it